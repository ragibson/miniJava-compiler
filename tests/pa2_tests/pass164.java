// PA1 parse assign pass
class Test {

    void p(int a) {
        Test [ a + 1]  = a * 3;
    }
}

