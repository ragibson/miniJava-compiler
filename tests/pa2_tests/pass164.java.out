********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [1]
  .   .   . ParameterDecl
  .   .   .   INT BaseType
  .   .   .   "a"parametername 
  .   .   StmtList [1]
  .   .   . AssignStmt
  .   .   .   IxRef
  .   .   .     IdRef
  .   .   .       "Test" Identifier
  .   .   .     BinaryExpr
  .   .   .       "+" Operator
  .   .   .         RefExpr
  .   .   .           IdRef
  .   .   .             "a" Identifier
  .   .   .         LiteralExpr
  .   .   .           "1" IntLiteral
  .   .   .   BinaryExpr
  .   .   .     "*" Operator
  .   .   .       RefExpr
  .   .   .         IdRef
  .   .   .           "a" Identifier
  .   .   .       LiteralExpr
  .   .   .         "3" IntLiteral
=============================================
